src/player/Player.php

```diff
@@ -946,6 +949,36 @@ public function stopSleep() : void{
 		}
 	}
 
+	public function setSneaking(bool $value = true) : void{
+		parent::setSneaking($value);
+
+		if($value){
+			$this->setSize($this->getInitialSizeInfo()->multiply(11 / 12,1,11 / 12)->scale($this->getScale()));
+		}else{
+			$this->resetSize();
+		}
+	}
+
+	public function setGliding(bool $value = true) : void{
+		parent::setGliding($value);
+
+		if($value){
+			$this->setSize((new EntitySizeInfo($this->size->getWidth(),$this->size->getWidth(),$this->size->getWidth()-($this->size->getHeight() - $this->size->getEyeHeight())))->scale($this->getScale()));
+		}else{
+			$this->resetSize();
+		}
+	}
+
+	public function setSwimming(bool $value = true) : void{
+		parent::setSwimming($value);
+
+		if($value){
+			$this->setSize((new EntitySizeInfo($this->size->getWidth(),$this->size->getWidth(),$this->size->getWidth()-($this->size->getHeight() - $this->size->getEyeHeight())))->scale($this->getScale()));
```

Comment:
I don't think this logic should be in Player, as it doesn't apply to just players, for example, Drowned. They may not be implemented right now, but if something like that is implemented, they will require the same size logic

Reply:
`Living` would be better.

Reply:
Already resolved locally